DigitalLearningSolutions.Data.Tests/DataServices/SupervisorDelegateDataServiceTests.cs

```diff
@@ -0,0 +1,123 @@
+                CandidateID = 254480,
+                Added = DateTime.Parse("2021-06-28 16:40:35.507"),
+                NotificationSent = DateTime.Parse("2021-06-28 16:40:35.507"),
+                Removed = null,
+                SupervisorEmail = "kevin.whittaker@hee.nhs.uk",
+                AddedByDelegate = false,
+                CentreId = 101,
+            };
+
+            // When
+            var result =
+                supervisorDelegateDataService.GetSupervisorDelegateRecordByInviteHash(
+                    inviteHashForFirstSupervisorDelegateRecord
+                );
+
+            // Then
+            result.Should().BeEquivalentTo(expectedRecord);
+        }
+
+        [Test]
+        public void GetPendingSupervisorDelegateRecordsByEmailsAndCentre_returns_correct_records()
+        {
+            using var transaction = new TransactionScope();
+
+            // Given
+            const int centreId = 101;
+            const string delegateEmailForValidRecord = "primary@email.com";
+            const string delegateEmailForRecordWithNonNullCandidateId = "kevin.whittaker@hee.nhs.uk";
+            const string delegateEmailForRemovedRecord = "louis.theroux@gmail.com";
+            const string? nullDelegateEmail = null;
+            var currentTime = DateTime.UtcNow;
+
+            connection.Execute(
+                @"INSERT INTO SupervisorDelegates (SupervisorAdminID, DelegateEmail, CandidateID, Added,
+                                 NotificationSent, Removed, SupervisorEmail, AddedByDelegate, InviteHash)
+                    OUTPUT INSERTED.ID
+                    VALUES (1, @delegateEmailForValidRecord, null, @currentTime, @currentTime, null,
+                            'supervisor@email.com', 0, '72e44c4d-77bd-4bed-a254-7cc27ab32928')",
+                new { delegateEmailForValidRecord, currentTime }
+            );
+
+            // When
+            var result = supervisorDelegateDataService.GetPendingSupervisorDelegateRecordsByEmailsAndCentre(
+                centreId,
+                new List<string?>
+                {
+                    delegateEmailForValidRecord,
+                    delegateEmailForRecordWithNonNullCandidateId,
+                    delegateEmailForRemovedRecord,
+                    nullDelegateEmail,
+                }
+            ).ToList();
+
+            // Then
+            result.Count.Should().Be(1);
+            result.First().DelegateEmail.Should().Be(delegateEmailForValidRecord);
+        }
+
+        [Test]
+        public void UpdateSupervisorDelegateRecordsCandidateId_updates_record_correctly()
+        {
+            using var transaction = new TransactionScope();
+
+            // Given
+            const int candidateId = 1;
+
+            // When
+            supervisorDelegateDataService.UpdateSupervisorDelegateRecordsCandidateId(
+                new List<int> { 6, 7, 8 },
+                1
+            );
+
+            // Then
+            var updatedRecord =

```

Comment:
This isn't explicitly testing that the candidateId has been changed, just that it is 1 after the update method has been called...it could have been 1 beforehand too. To remedy this:
* `var oldRecord = supervisorDelegateDataService.GetSupervisorDelegateRecordByInviteHash(inviteHashForFirstSupervisorDelegateRecord)` before updating
* Test that `oldRecord.CandidateID` is not 1
* Test that `oldRecord.ID` is in the list [6,7,8] (otherwise you aren't explicitly proving that the record you're updating is the same as the record you're getting from the database after the update)

Reply:
Thanks, made those changes